From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Alastair D'Silva <alastair@d-silva.org>
Date: Thu, 13 Aug 2026 00:00:00 +0000
Subject: orangepi4pro: enable PCIe power GPIO (PL3) and disable pd_pcie_test

On the Orange Pi 4 Pro, power to the M.2 PCIe slot is gated by GPIO PL3 (r_pio).
This patch configures power-gpios on &pcie_rc to assert PL3, ensures the PCIe
power domain is marked always-on, and disables mock power domain test nodes
to prevent unwanted power cycling of the NVMe interface.

Signed-off-by: Alastair D'Silva <alastair@d-silva.org>
Assisted-by: Antigravity <antigravity@google.com>

--- a/arch/arm64/boot/dts/allwinner/sun60i-a733-orangepi-4-pro.dts
+++ b/arch/arm64/boot/dts/allwinner/sun60i-a733-orangepi-4-pro.dts
@@ -1651,8 +1651,26 @@
 &pcie_rc {
-	//power-gpios =  <&r_pio PL 3 GPIO_ACTIVE_HIGH>;
+	power-gpios = <&r_pio PL 3 GPIO_ACTIVE_HIGH>;
 	reset-gpios = <&pio PD 22 GPIO_ACTIVE_HIGH>;
 	wake-gpios = <&pio PD 21 GPIO_ACTIVE_HIGH>;
 	pcie1v8-supply = <&reg_bldo1>;
 	pcie3v3-supply = <&reg_dcdc1>;
 	status = "okay";
 };
+
+&pd {
+	pd_pcie@SUN60IW2_PCK_PCIE {
+		ppu-always-on;
+	};
+};
+
+&pd_vi_test { status = "disabled"; };
+&pd_ve_dec_test { status = "disabled"; };
+&pd_ve_enc_test { status = "disabled"; };
+&pd_npu_test { status = "disabled"; };
+&pd_gpu_top_test { status = "disabled"; };
+&pd_gpu_core_test { status = "disabled"; };
+&pd_pcie_test { status = "disabled"; };
+&pd_usb2_test { status = "disabled"; };
+&pd_de_sys_test { status = "disabled"; };
+&pd_vo_test { status = "disabled"; };
+&pd_vo1_test { status = "disabled"; };
